### ISSUE TITLE
Head API: Link onload event handler is not being called when passing a function

### ISSUE THREAD

**Issue by duydao** [gatsbyjs/gatsby#37345]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

It seems like the onload event handler is not beeing triggered when using the Head API:

```
import * as React from "react"

const IndexPage = () => {
  function callback() {
    console.log("this works as expected")
  }
  return (
      <link
          onLoad={callback}
          rel="preload"
          as="style"
          href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css"
      />
  )
}

export function Head({ data }) {
  function callback() {
    console.log("this does not get called")
  }

  return (
      <link
          onLoad={callback}
          rel="preload"
          as="style"
          href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css"
      />
  )
}

export default IndexPage
```

When using the same code as component body, it works as expected.

### Reproduction Link

https://stackblitz.com/edit/github-vhndv9?file=src%2Fpages%2Findex.js

### Steps to Reproduce

1. Open index page
2. check web console
...

### Expected Result

There should be two console messages:
- this works as expected
- this does not get called

### Actual Result

Only one message is shown
- this works as expected

### Environment

```shell
Mac OS 13.1
node v18.12.1
Firefox 108.0.1
Gatsby 5.3.3
React 18.2
```

### Config Flags

I don't see any flags.

**Comment by marvinjude** [comment]
Hi @duydao,

Thanks for reporting this.  This is happening because we use [node.cloneNode()](https://developer.mozilla.org/en-US/docs/Web/API/Node/cloneNode#:~:text=Cloning%20a%20node%20copies%20all%20of%20its%20attributes%20and%20their%20values%2C%20including%20intrinsic%20(inline)%20listeners.%20It%20does%20not%20copy%20event%20listeners%20added%20using%20addEventListener()%20or%20those%20assigned%20to%20element%20properties) in [head-export-handler-for-browser.js#L36 ](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/cache-dir/head/head-export-handler-for-browser.js#L36) which doesn't copy along even listeners. The fix here will be to somehow copy along event listeners.

**Comment by marvinjude** [comment]
Hey @duydao,

Implementing this is kind of hairy right now because of [how react handles event listeners](https://dev.to/romaintrotard/under-the-hood-of-event-listeners-in-react-4g01). Can you explain your use-case?
For context, given a code like this:
```js
export function Head() {
  return (
    <link
      onError={() => console.log("e")}
      onLoad={() => console.log("888")}
      href="https://www.canonical.ie/"
    />
  );
}
```

The resulting react node looks like the below:
<img width="1243" alt="Screenshot 2023-01-17 at 09 43 26" src="https://user-images.githubusercontent.com/17142206/212851054-96481684-4ebb-4dc1-8dec-e2358a3daba7.png">

Basically, the event handlers are behind some random hash. Trying to extract those may be end up being problematic. We would appreciate some explanation as to why you want to use event listeners in `Head`

**Comment by duydao** [comment]
Hi @marvinjude,
thanks for getting back to me.

We are using gatsby to deliver "legacy" sites. Basically what we're doing is fetch the site, convert the header and part of the body and deliver the site by using SSG or SSR.

The particular code we're trying to convert is described in this stackoverflow:
https://stackoverflow.com/a/65205954

`<link rel="preload" as="style" onload="this.onload=null;this.rel='stylesheet'" href="http://www.example.com/style.css"/>`

the preload value of the rel atrribute is converted to stylesheet in order to apply it to the document.

The workaround we're using right now is to render the links as stylesheet instead of preload.

I think this is a special use case that isn't very common. I will close this issue. Thanks!
